client/pd_service_discovery.go

```diff
@@ -97,6 +100,237 @@ type ServiceDiscovery interface {
 	AddServiceAddrsSwitchedCallback(callbacks ...func())
 }
 
+// ServiceClient is an interface that defines a set of operations for a raw PD gRPC client to specific PD server.
+type ServiceClient interface {
+	// GetAddress returns the address information of the PD server.
+	GetAddress() string
+	// GetClientConn returns the gRPC connection of the service client
+	GetClientConn() *grpc.ClientConn
+	// BuildGRPCContext builds a context object with a gRPC context.
+	// ctx: the original context object.
+	// mustLeader: whether must send to leader.
+	BuildGRPCContext(ctx context.Context, mustLeader bool) context.Context
+	// IsLeader returns whether the target PD server is leader.
+	IsLeader() bool
```

Comment:
What if the leader changes?

Reply:
If leader changes, it will create new ServiceClient and store it in `leader`

Reply:
How about targetIsLeader or something else?